### PR TITLE
Improve IAP setup process

### PR DESCRIPTION
**Pull request by blackzlq**
Design doc: https://docs.google.com/document/d/1aubk94Pd6qt1G2B2e-awOQ2ranzwgnSLikj5I6fBUwY/edit?resourcekey=0-2d-BsBiGOkwWrGb5aDYDTg&tab=t.0

1. Change the authenticator to auto fetch the backend service
2. add related permissions

Tested-by: zlq

Tested:
Sandbox

### CODE REVIEW

**Comment by blackzlq** [comment]
rebase

File: modules/jupyter/authentication/authenticator/gcpiapjwtauthenticator/gcpiapjwtauthenticator.py [hunk 1]
```diff
@@ -22,34 +22,70 @@
...
-        expected_audience = self.authenticator.expected_audience
-    
+
+        # Extract project ID, namespace, and backend config name from the authenticator
+        project_id = self.authenticator.project_id
+        namespace = self.authenticator.namespace
+        service_name = self.authenticator.service_name
+        print("Project ID:", project_id)
+        print("Namespace:", namespace)
+        print("Backend Config Name:", service_name)
```
Comment (chiayi) [inline modules/jupyter/authentication/authenticator/gcpiapjwtauthenticator/gcpiapjwtauthenticator.py:55]: NIT: Change `backend config name` avoid confusion with the backend config deployment?

File: modules/jupyter/authentication/authenticator/gcpiapjwtauthenticator/gcpiapjwtauthenticator.py [hunk 2]
```diff
@@ -59,33 +95,48 @@ class GCPIAPAuthenticator(Authenticator):
...
+        default_value='',
+        config=True,
+        help="""Expected namespace""")
+
+    service_name = Unicode(
         default_value='',
         config=True,
-        help="""Expected Audience of the authenication JWT""")
-    
+        help="""Expected backend_config_name""")
```
Comment (chiayi) [inline modules/jupyter/authentication/authenticator/gcpiapjwtauthenticator/gcpiapjwtauthenticator.py:125]: Same here with the backend config name

File: applications/jupyter/main.tf [hunk 3]
```diff
@@ -71,9 +71,8 @@ module "jupyterhub" {
  project_id = var.project_id
 
  namespace = var.namespace
- gcp_service_account = var.gcp_service_account
+ gcp_and_k8s_service_account = var.gcp_and_k8s_service_account
```
Comment (chiayi) [inline applications/jupyter/main.tf:74]: Should update variable name update on README.md

File: modules/jupyter/variables.tf [hunk 4]
```diff
@@ -23,9 +23,10 @@ variable "default_backend_service" {
 }
 
 variable "members_allowlist" {
-  type        = string
-  default     = ""
+  type    = list(string)
```
Comment (chiayi) [inline modules/jupyter/variables.tf:26]: Do we still need memebrs_allowlist if we are removing `google_iap_web_backend_service_iam_binding`?

**Pull request closed without merging**